src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -127,6 +159,34 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     this
   }
 
+  /*
+   * Implemented by means of a buffer to keep the last n elements from being returned during iteration.
+   */
+  def dropRight(n: Int): Iterator[A] = {
```

Comment:
I think that if we want to avoid buffering we have to perform two traversals. I’m not sure which solution is the best in terms of execution times.